File: packages/js/core/src/uri-resolution/core/types/UriResolver.ts
import { WrapperCache, Client, Uri } from "../../../types";
import { UriResolutionResult, UriResolutionStack } from ".";

import { DeserializeManifestOptions } from "@polywrap/wrap-manifest-types-js";

export abstract class UriResolver {
  public abstract get name(): string;

  public abstract resolveUri(
    uri: Uri,
    client: Client,
    cache: WrapperCache,
    resolutionPath: UriResolutionStack,
    deserializeOptions?: DeserializeManifestOptions
  ): Promise<UriResolutionResult>;
}

File: packages/py/core/core/uri_resolution/core/types/uri_resolver.py
from __future__ import annotations

from abc import ABC, abstractmethod

from ....types import Client, Uri, WrapperCache
from .uri_resolution_result import UriResolutionResult
from .uri_resolution_stack import UriResolutionStack


class UriResolver(ABC):
    @property
    @abstractmethod
    def name(self) -> str:
        pass

    @abstractmethod
    async def resolve_uri(
        self, uri: Uri, client: Client, cache: WrapperCache, resolution_path: UriResolutionStack
    ) -> UriResolutionResult:
        pass
